editor.planx.uk/src/@planx/components/Pay/Editor.tsx

```diff
@@ -281,7 +314,10 @@ const Component: React.FC<Props> = (props: Props) => {
                       setFieldValue("govPayMetadata", metadata);
                     }}
                     Editor={GovPayMetadataEditor}
-                    newValue={() => ({ key: "", value: "" })}
+                    newValue={() => {
+                      setTouched({});
+                      return { key: "", value: "" };
+                    }}
```

Comment:
This is a bit of an awkward workaround of this issue - https://github.com/jaredpalmer/formik/issues/1616

As pointed out by Silvia when testing (see comments here), if 
 - a field has an error
 - you delete it
 - and add a new field
 
The error persists, by array index presumably (?), on the new field.

This code change forces a reset of the touched status when adding a new field, and error display is now correctly tied to status also (an error will only display when a field is touched).